packages/api/src/hooks/middleware/createDefaultGroupActivitiesMiddleware.ts

```diff
@@ -25,8 +26,19 @@ function bin<T>(items: T[], grouping: (last: T, current: T) => boolean): T[][] {
   return bins;
 }
 
-function sending(activity: DirectLineActivity): boolean {
-  return activity.from.role === 'user' && activity.channelData && activity.channelData.state !== SENT;
+function sending(activity) {
+  if (activity.from.role === 'user') {
```

Comment:
In previous cases, we will see 2 timestamps on 2 activities which are supposed to be collapsed together.

This is because some activity could have `state` of `undefined`, while the other one has `state` of `"sent"`. Both convey the same meaning: the message has been sent.

The `state` was missing sometimes because it was received from ABS. Then we immediately patch it with `"sent"`. This patching is done by Redux Saga and is, unfortunately, async operation. It cause a render and flash.